pkg/credentialprovider/plugin/framework/plugin.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/kubernetes/pkg/credentialprovider/apis/criauth"
+	"k8s.io/kubernetes/pkg/credentialprovider/apis/criauth/install"
+	"k8s.io/kubernetes/pkg/credentialprovider/apis/criauth/v1alpha1"
+)
+
+var (
+	scheme = runtime.NewScheme()
+	codecs = serializer.NewCodecFactory(scheme)
+)
+
+func init() {
+	install.Install(scheme)
+}
+
+// CRIAuthPlugin is an interface implemented by the CRI auth plugin to fetch
+// the username/password based on the provided image name.
+type CRIAuthPlugin interface {
+	GetAuthPluginResponse(image string, extraArgs map[string]string) (response *v1alpha1.CRIAuthPluginResponse, err error)
```

Comment:
I agree it would be more convenient but it would probably be harder to test. The interface allows us to create "fake" implementations of the plugin for better testability. I think it would be hard to accomplish this with what you proposed.